Description: Build a shared library properly.
 - build the library as libsdb.so.0.0.0
 - pass the -soname parameter to the linker with version 0
 - break out the compiler and linker flags for the shared build
Forwarded: not yet
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2013-03-27

--- a/config.mk
+++ b/config.mk
@@ -8,6 +8,7 @@
 CFLAGS+=-Wall
 #CFLAGS+=-O3
 #CFLAGS+=-ggdb -g -Wall -O0
+CFLAGS_SHARED?=-fPIC -shared
 
 HAVE_VALA=$(shell valac --version)
 # This is hacky
@@ -20,7 +21,9 @@
    CC+=-arch i386 -arch x86_64
  endif
 else
-SOEXT=so
+SOVERSION=0
+SOEXT=so.0.0.0
+LDFLAGS_SHARED+=-Wl,-soname,libsdb.so.$(SOVERSION)
 endif
 RANLIB?=ranlib
 EXEXT=
--- a/src/Makefile
+++ b/src/Makefile
@@ -29,7 +29,7 @@
 	${RANLIB} libsdb.a
 
 libsdb.${SOEXT}: ${OBJ}
-	${CC} ${CFLAGS} ${LDFLAGS} -o $@ $(subst .o,.c,${OBJ}) -fPIC -shared
+	${CC} ${CFLAGS} $(CFLAGS_SHARED) ${LDFLAGS} $(LDFLAGS_SHARED) -o $@ $(subst .o,.c,${OBJ})
 
 t:
 	rm -f foo
